diff --git a/components/email.inc b/components/email.inc
index 6430843da..7ba3dbeb1 100644
--- a/components/email.inc
+++ b/components/email.inc
@@ -152,7 +152,7 @@ function _webform_render_email($component, $value = NULL, $filter = TRUE, $submi
     '#required' => $component['required'],
     '#weight' => $component['weight'],
     '#description' => $filter ? webform_filter_descriptions($component['extra']['description'], $node) : $component['extra']['description'],
-    '#attributes' => $component['extra']['attributes'],
+    '#attributes' => $component['extra']['attributes'] ?: array(),
     '#element_validate'  => array('_webform_validate_email'),
     '#theme_wrappers' => array('webform_element'),
     '#translatable' => array('title', 'description', 'placeholder'),
diff --git a/components/file.inc b/components/file.inc
index b8233e507..3ae2bc044 100644
--- a/components/file.inc
+++ b/components/file.inc
@@ -347,7 +347,7 @@ function _webform_render_file($component, $value = NULL, $filter = TRUE, $submis
     '#title_display' => $component['extra']['title_display'] ? $component['extra']['title_display'] : 'before',
     '#required' => $component['required'],
     '#default_value' => isset($value[0]) ? $value[0] : NULL,
-    '#attributes' => $component['extra']['attributes'],
+    '#attributes' => $component['extra']['attributes'] ?: array(),
     '#upload_validators' => array(
       'file_validate_size' => array($max_filesize),
       'file_validate_extensions' => array(implode(' ', $component['extra']['filtering']['types'])),
diff --git a/components/number.inc b/components/number.inc
index 46a952953..ab3031231 100644
--- a/components/number.inc
+++ b/components/number.inc
@@ -295,7 +295,7 @@ function _webform_render_number($component, $value = NULL, $filter = TRUE, $subm
     '#field_prefix' => empty($component['extra']['field_prefix']) ? NULL : ($filter ? webform_filter_xss($component['extra']['field_prefix']) : $component['extra']['field_prefix']),
     '#field_suffix' => empty($component['extra']['field_suffix']) ? NULL : ($filter ? webform_filter_xss($component['extra']['field_suffix']) : $component['extra']['field_suffix']),
     '#description' => $filter ? webform_filter_descriptions($component['extra']['description'], $node) : $component['extra']['description'],
-    '#attributes' => $component['extra']['attributes'],
+    '#attributes' => $component['extra']['attributes'] ?: array(),
     '#element_validate' => array('_webform_validate_number'),
     '#theme_wrappers' => array('webform_element'),
     '#min' => $component['extra']['min'],
diff --git a/components/textarea.inc b/components/textarea.inc
index 7a173d41f..2bfe6e132 100644
--- a/components/textarea.inc
+++ b/components/textarea.inc
@@ -119,7 +119,7 @@ function _webform_render_textarea($component, $value = NULL, $filter = TRUE, $su
     '#description' => $filter ? webform_filter_descriptions($component['extra']['description'], $node) : $component['extra']['description'],
     '#rows' => !empty($component['extra']['rows']) ? $component['extra']['rows'] : 5,
     '#cols' => !empty($component['extra']['cols']) ? $component['extra']['cols'] : 60,
-    '#attributes' => $component['extra']['attributes'],
+    '#attributes' => $component['extra']['attributes'] ?: array(),
     // MUST be FALSE to disable.
     '#resizable' => (bool) $component['extra']['resizable'],
     '#theme_wrappers' => array('webform_element'),
diff --git a/components/textfield.inc b/components/textfield.inc
index d7bb1a3d8..48f05f1e7 100644
--- a/components/textfield.inc
+++ b/components/textfield.inc
@@ -156,7 +156,7 @@ function _webform_render_textfield($component, $value = NULL, $filter = TRUE, $s
     '#field_prefix' => empty($component['extra']['field_prefix']) ? NULL : ($filter ? webform_filter_xss($component['extra']['field_prefix']) : $component['extra']['field_prefix']),
     '#field_suffix' => empty($component['extra']['field_suffix']) ? NULL : ($filter ? webform_filter_xss($component['extra']['field_suffix']) : $component['extra']['field_suffix']),
     '#description' => $filter ? webform_filter_descriptions($component['extra']['description'], $node) : $component['extra']['description'],
-    '#attributes' => $component['extra']['attributes'],
+    '#attributes' => $component['extra']['attributes'] ?: array(),
     '#theme_wrappers' => array('webform_element'),
     '#translatable' => array(
       'title',
--
2.34.1+GitX
